src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts

```diff
@@ -3,86 +3,184 @@
+import { NotebookServiceStub } from 'sql/workbench/contrib/notebook/test/electron-browser/common';
 
 suite('notebookUtils', function (): void {
-	let conn: IConnectionProfile = {
-		connectionName: '',
-		serverName: '',
-		databaseName: '',
-		userName: '',
-		password: '',
-		authenticationType: '',
-		savePassword: true,
-		groupFullName: '',
-		groupId: '',
-		providerName: mssqlProviderName,
-		saveProfile: true,
-		id: '',
-		options: {},
-		azureTenantId: undefined,
-		azureAccount: undefined
+	const mockNotebookService = TypeMoq.Mock.ofType<INotebookService>(NotebookServiceStub);
+	const defaultTestProvider = 'testDefaultProvider';
+	const testProvider = 'testProvider';
+	const testKernel: nb.IStandardKernel = {
+		name: 'testName',
+		displayName: 'testDisplayName',
+		connectionProviderIds: ['testId1', 'testId2']
 	};
 
-	test('Should format server and database name correctly for attach to', async function (): Promise<void> {
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = 'serverName';
-		connProfile.databaseName = 'databaseName';
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, 'serverName (databaseName)');
+	function setupMockNotebookService() {
+		mockNotebookService.setup(n => n.getProvidersForFileType(TypeMoq.It.isAnyString()))
+			.returns((fileName, service) => {
+				if (fileName === DEFAULT_NOTEBOOK_FILETYPE) {
+					return [defaultTestProvider];
+				} else {
+					return [testProvider];
+				}
+			});
+
+		// getStandardKernelsForProvider
+		mockNotebookService.setup(n => n.getStandardKernelsForProvider(TypeMoq.It.isAnyString()))
+			.returns((provider) => {
+				return [testKernel];
+			});
+	}
+
+	test('isStream Test', async function (): Promise<void> {
+		let result = isStream(<nb.ICellOutput>{
+			output_type: 'stream'
+		});
+		assert.strictEqual(result, true);
+
+		result = isStream(<nb.ICellOutput>{
+			output_type: 'display_data'
+		});
+		assert.strictEqual(result, false);
+
+		result = isStream(<nb.ICellOutput>{
+			output_type: undefined
+		});
+		assert.strictEqual(result, false);
 	});
 
-	test('Should format server name correctly for attach to', async function (): Promise<void> {
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = 'serverName';
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, 'serverName');
+	test('getProvidersForFileName Test', async function (): Promise<void> {
+		setupMockNotebookService();
+
+		let result = getProvidersForFileName('', mockNotebookService.object);
+		assert.deepStrictEqual(result, [defaultTestProvider]);
```

Comment:
are we positive that deepStrictEqual will return true only if 2 arrays are have identical contents? just sanity checking 😄

Reply:
Yes. Changing the contents of the expected array here fails the test.